### PR TITLE
🚩 Add progress_bar to filters/algorithms

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add progress_bar to filters/algorithms

 
relate to https://github.com/pyvista/pyvista/issues/1518

resolves https://github.com/pyvista/pyvista/issues/1518

### Details

- None

### CODE REVIEW

File: pyvista/core/filters/poly_data.py [hunk 1]
```diff
@@ -662,7 +691,7 @@ def smooth(poly_data, n_iter=20, relaxation_factor=0.01, convergence=0.0,
         alg.SetEdgeAngle(edge_angle)
         alg.SetBoundarySmoothing(boundary_smoothing)
         alg.SetRelaxationFactor(relaxation_factor)
-        alg.Update()
+        _update_alg(alg, progress_bar, 'Smoothing PolyData')
```
Comment (tkoyama010) [inline pyvista/core/filters/poly_data.py:694]:
```suggestion
        _update_alg(alg, progress_bar, 'Smoothing Mesh')
```

File: pyvista/core/filters/poly_data.py [hunk 2]
```diff
@@ -926,7 +964,7 @@ def subdivide(poly_data, nsub, subfilter='linear', inplace=False):
         # Subdivide
         sfilter.SetNumberOfSubdivisions(nsub)
         sfilter.SetInputData(poly_data)
-        sfilter.Update()
+        _update_alg(sfilter, progress_bar, 'Subdividing')
```
Comment (tkoyama010) [inline pyvista/core/filters/poly_data.py:967]:
```suggestion
        _update_alg(sfilter, progress_bar, 'Subdividing Mesh')
```

File: pyvista/core/filters/poly_data.py [hunk 3]
```diff
@@ -1017,7 +1059,7 @@ def subdivide_adaptive(poly_data, max_edge_len=None, max_tri_area=None,
             sfilter.SetMaximumNumberOfPasses(max_n_passes)
 
         sfilter.SetInputData(poly_data)
-        sfilter.Update()
+        _update_alg(sfilter, progress_bar, 'Subdividing Adaptive')
```
Comment (tkoyama010) [inline pyvista/core/filters/poly_data.py:1062]:
```suggestion
        _update_alg(sfilter, progress_bar, 'Subdividing Mesh Adaptive')
```

File: pyvista/core/filters/poly_data.py [hunk 4]
```diff
@@ -1142,7 +1184,7 @@ def decimate(poly_data, target_reduction, volume_preservation=False,
         alg.SetTargetReduction(target_reduction)
 
         alg.SetInputData(poly_data)
-        _update_alg(alg, progress_bar, 'Decimating')
+        _update_alg(alg, progress_bar, 'Decimating Quadric')
```
Comment (tkoyama010) [inline pyvista/core/filters/poly_data.py:1187]:
```suggestion
        _update_alg(alg, progress_bar, 'Decimating Mesh')
```

File: pyvista/core/filters/data_set.py [hunk 5]
```diff
@@ -2445,12 +2537,13 @@ def streamlines(dataset, vectors=None, source_center=None,
...
+            _update_alg(source, progress_bar, 'Making Line')
         else:
             source = _vtk.vtkPointSource()
             source.SetCenter(source_center)
             source.SetRadius(source_radius)
             source.SetNumberOfPoints(n_points)
-        source.Update()
+            _update_alg(source, progress_bar, 'Making Point')
         input_source = pyvista.wrap(source.GetOutput())
         output = dataset.streamlines_from_source(input_source, vectors, **kwargs)
```
Comment (akaszynski) [inline pyvista/core/filters/data_set.py:2548]: Pass the progress bar arg here as well.

File: pyvista/core/filters/data_set.py [hunk 6]
```diff
@@ -2445,12 +2537,13 @@ def streamlines(dataset, vectors=None, source_center=None,
             source.SetPoint1(pointa)
             source.SetPoint2(pointb)
             source.SetResolution(n_points)
+            _update_alg(source, progress_bar, 'Making Line')
```
Comment (akaszynski) [inline pyvista/core/filters/data_set.py:2540]: We don't need a progress bar here.  It's just a single line and will compute instantly.

File: pyvista/core/filters/data_set.py [hunk 7]
```diff
@@ -2445,12 +2537,13 @@ def streamlines(dataset, vectors=None, source_center=None,
...
         else:
             source = _vtk.vtkPointSource()
             source.SetCenter(source_center)
             source.SetRadius(source_radius)
             source.SetNumberOfPoints(n_points)
-        source.Update()
+            _update_alg(source, progress_bar, 'Making Point')
```
Comment (akaszynski) [inline pyvista/core/filters/data_set.py:2546]: Same here,  Revert this please.

File: pyvista/core/filters/poly_data.py [hunk 8]
```diff
@@ -53,14 +56,14 @@ def edge_mask(poly_data, angle):
         featureEdges.NonManifoldEdgesOff()
         featureEdges.ManifoldEdgesOff()
         featureEdges.SetFeatureAngle(angle)
-        featureEdges.Update()
+        _update_alg(featureEdges, progress_bar, 'Edging Mask')
```
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:59]:
```suggestion
        _update_alg(featureEdges, progress_bar, 'Computing Edge Mask')
```

File: pyvista/core/filters/poly_data.py [hunk 9]
```diff
@@ -80,7 +83,7 @@ def _boolean(poly_data, btype, other_mesh, tolerance):
         bfilter.SetInputData(1, other_mesh)
         bfilter.ReorientDifferenceCellsOn()  # this is already default
         bfilter.SetTolerance(tolerance)
-        bfilter.Update()
+        _update_alg(bfilter, progress_bar, 'Boolean Operating')
```
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:86]:
```suggestion
        _update_alg(bfilter, progress_bar, 'Performing Boolean Operation')
```

File: pyvista/core/filters/poly_data.py [hunk 10]
```diff
@@ -503,7 +525,7 @@ def curvature(poly_data, curv_type='mean'):
         else:
             raise ValueError('``curv_type`` must be either "Mean", '
                              '"Gaussian", "Maximum", or "Minimum".')
-        curvefilter.Update()
+        _update_alg(curvefilter, progress_bar, 'Computing Curvatures')
```
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:528]:
```suggestion
        _update_alg(curvefilter, progress_bar, 'Computing Curvature')
```

File: pyvista/core/filters/poly_data.py [hunk 11]
```diff
@@ -1017,7 +1059,7 @@ def subdivide_adaptive(poly_data, max_edge_len=None, max_tri_area=None,
             sfilter.SetMaximumNumberOfPasses(max_n_passes)
 
         sfilter.SetInputData(poly_data)
-        sfilter.Update()
+        _update_alg(sfilter, progress_bar, 'Subdividing Mesh Adaptive')
```
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:1062]:
```suggestion
        _update_alg(sfilter, progress_bar, 'Adaptively Subdividing Mesh')
```

**Review by akaszynski — CHANGES_REQUESTED** [review]
Thanks for adding this!

Looks good so far.  Only request would be to not have parameterized tests for the progress bar.  Reason being is we have coverage already for the progress bar, and we we actually need to test if a keyword can be set, we only need to set it and not test both states.  Our unit testing is quite extensive as it is and I don't want it to bloat.

So, let's modify the unit tests to just set ``progress_bar=True`` (or ``False``) for the methods.

File: pyvista/core/filters/data_set.py (hunk 6 above)
Comment (tkoyama010) [reply pyvista/core/filters/data_set.py:2540]:
```suggestion
```

File: pyvista/core/filters/data_set.py (hunk 7 above)
Comment (tkoyama010) [reply pyvista/core/filters/data_set.py:2546]:
```suggestion
        source.Update()
```

File: pyvista/core/filters/data_set.py (hunk 5 above)
Comment (tkoyama010) [reply pyvista/core/filters/data_set.py]: Fix in https://github.com/pyvista/pyvista/pull/1519/commits/22305b87045b5da21cf7fc19af405154fc2bd75d

**Review by tkoyama010 — COMMENTED** [review]
@akaszynski Thanks for your review. I set `progress_bar=True` in test.

**Review by akaszynski — APPROVED** [review]
CI is still running, but tests and style passes locally.  In the future, I'd reduce the number of commits made, but it looks like you're making them using the web interface, and maybe that's harder to do.  Issue is that the CI runs for every commit, so it's still a bit backed up.

Regardless, thanks again for adding this!  We'll have to make sure to add the `progress_bar` parameter for all new filters.

**Pull request merged**